extension/searchdefault.js

```diff
@@ -0,0 +1,49 @@
+"use strict";
+
+let addonData;
+window.onload = async () => {
+  let params = new URLSearchParams(window.location.search);
+
+  addonData = await browser.management.get(params.get("id"));
+  for (let el of document.getElementsByClassName("addon-name")) {
+    el.appendChild(document.createTextNode(addonData.name));
+  };
+  let description = document.getElementsByClassName("addon-description");
+  let blocks = addonData.description.split("\n");
+  for (let tn of blocks) {
+    let div = document.createElement("div");
+    div.appendChild(document.createTextNode(tn));
+    description[0].appendChild(div);
+  }
+
+  // Add the disabled date
+  let date = await browser.searchDefaults.deactivatedDate(addonData.id);
+  let text = new Intl.DateTimeFormat('en-US', {month: "long", year: "numeric"}).format(new Date(date));
+  let dateEl = document.getElementsByClassName("addon-date");
+  dateEl[0].appendChild(document.createTextNode(text));
+
+  let icons = document.getElementsByClassName("icon");
+  let addonIcons = addonData.icons.sort((a, b) => { return a.size < b.size; });
```

Comment:
`addonData.icons` may be `undefined`, and `addonIcons[0]` may be too, which can break the logic here.

I suggest to replace `let addonIcons = ...` until `icons[0].setAttribute("src", iconSrc);` with:

```javascript
let addonIcons = addonData.icons || [];
addonIcons.sort((a, b) => b.size - a.size);
// Use size 48 icon, and otherwise the largest.
let addonIcon = addonIcons.find(a => a.size == 48) || addonIcons[0];
if (addonIcon) {
  icons[0].src = addonIcon.url;
}
```

Reply:
I've done this in https://github.com/mozilla-extensions/restore-search-defaults/pull/19/commits/a4f26c2b32e567a0bb440a13caa4ffbba0548416